main.go

```diff
@@ -0,0 +1,323 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/himanshuo/gameoflife/models"
+	_ "github.com/mattn/go-sqlite3"
+	"html/template"
+	"log"
+	"net/http"
+	"strconv"
+)
+
+//_ is in order to import a package solely for its side-effects at initialization.
+//In this case, go-sqlite3's side effects are allowing sqlite3 to be usable as a
+//database for  sql.Open
+
+//the database resource
+var db *sql.DB
+var Router *mux.Router
+
+//run at start of program.
+func init() {
+
+	startDB()
+	Router = mux.NewRouter()
+
+}
+
+func startDB() error {
+	var err error
+	//sqlite 3 database is stored in /data/data.db file
+	db, err = sql.Open("sqlite3", "./data/data.db")
+	if err != nil {
+		return err
+	}
+
+	//make sure we can actually query the database.
+	if err := db.Ping(); err != nil {
+		return err
+	}
+
+	//ASSUMPTION: table exists in database
+
+	//statement creates Task table which only has a PK id and name textfields
+	sqlStmt := "create table Task (id integer not null primary key autoincrement, name text);"
+	//run statement to create table. return object of Result type is not needed.
+	_, err = db.Exec(sqlStmt)
+	if err != nil {
+		return err
+	}
+	log.Printf("Database Started")
+	return nil
+}
+
+//Views
+// Home page view. For now, simply lists all the tasks one by one.
+func Home(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Home Screen Opened")
+	//get all tasks
+	rows, err := db.Query("select id, name from Task")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer rows.Close()
+
+	//list of tasks
+	tasks := []models.Task{}
+
+	//iterate through response
+	for rows.Next() {
+		var id int
+		var name string
+		//fill id and name variables with db output
+		rows.Scan(&id, &name)
+		//create new controller model for task
+		curTask := models.Task{Id: id, Name: name}
+		//add model to tasks list
+		tasks = append(tasks, curTask)
+	}
+
+	//choose which template to show user when this endpoint is called
+	t, err := template.ParseFiles("views/static/templates/base.html")
```

Comment:
is there a better way to get the template path than a straight relative file path? At the very least, I can add constants at the start of this file.